src/data/markdown/docs/05 Examples/01 Examples/24 distribute-workloads.md

```diff
@@ -0,0 +1,229 @@
+  http.get("https://test.k6.io/news.php");
+}
+```
+
+### Tip: reduce scenario boilerplate with `new`
+
+If your script uses multiple scenarios, consider making a function to construct a scenario object.
+
+For example, the following script makes the same test as the preceding section.
+But instead of a new scenario each time, it uses the `new` operator to construct new workloads based on arguments for the number of VUs and `exec` name.
+
+<Collapsible title="workload constructor" isOpen="" tag="">
+
+<CodeGroup labels={["workload-constructor.js"]} lineNumbers={[true]} showCopyButton={[true]}>
+
+
+```javascript
+import http from "k6/http";
+
+// workload constructor
+function Scenario(vus, exec) {
+  this.executor = "shared-iterations";
+  this.vus = vus;
+  this.exec = exec;
+  this.iterations = 100;
+}
+
+//create scenario object
+const contactWorkload = new Scenario(80, "contacts");
+const newsWorkload = new Scenario(20, "news");
+
+export const options = {
+  //scenario to view contacts
+  scenarios: {
+    contactWorkload,
+    newsWorkload,
+  },
+};
+
+//use the exec property to run different scenarios for different functions
+
+export function contacts() {
+  http.get("https://test.k6.io/contacts.php");
+}
+
+export function news() {
+  http.get("https://test.k6.io/news.php");
+}
+
+```
+</CodeGroup>
+
+</Collapsible>
+
+
+
+
+## Configuring VUs based on VU id
+
+In some cases, it might be inconvenient or impractical to write a scenario for each behavior.
+As an alternative, you can distribute logic across a range of VUs with the [execution context variables](/using-k6/execution-context-variables) from the [`k6/execution`](https://k6.io/docs/javascript-api/k6-execution/) module.
+With the `exec` object, you can scope logic to a specific instance, scenario, or across all VUs.
+
+For example, this statement assigns behavior to the first 25 VUs in a test.
+
+```bash
+if (exec.vu.idInTest <= 25) {
+   //do something;
+  }
+```
+
+
+For more flexibility, you can use modulo expressions distribute VUs according to percentages.
+For example, the following script distributes logic according to different user profiles:
+- 40 percent of users check the news.
+- 60 percent play a coinflip game.
+   - Half bet `heads` and half bet `tails`.
+
+<CodeGroup labels={["behavior-based-on-exec-context.js"]} lineNumbers={[true]} showCopyButton={[true]}>
+
+```javascript
+import http from "k6/http";
+import exec from "k6/execution";
+
+export const options = {
+  scenarios: {
+    quickRamp: {
+      executor: "ramping-arrival-rate",
+      startRate: 0,
+      timeUnit: "1s",
+      preAllocatedVUs: 100,
+      stages: [
+        { target: 10, duration: "10s" },
+        { target: 10, duration: "15s" },
+        { target: 0, duration: "5s" },
+      ],
+    },
+  },
+};
+
+export default function () {
+  if (exec.vu.idInTest % 10 < 4) {
+    // 0-3 range, read the news
+    http.get("http://test.k6.io/news");
+  }
+  else if (exec.vu.idInTest % 10 < 7) {
```

Comment:
```suggestion
  } else if (exec.vu.idInTest % 10 < 7) {
```

I guess this is what you mean as the other `else` is similarly formatted, and *I* prefer it this way ;P